apkid/rules/elf/obfuscators.yara

```diff
@@ -155,6 +155,55 @@ rule ollvm : obfuscator
     not ollvm_v6_0_strenc
 }
 
+rule byteguard_0_9_3 : obfuscator
+{
+  meta:
+    description = "ByteGuard 0.9.3"
+    sample      = "eed4f7b907fe2173935d307dfb0d6aa7098f69db8dfb65e49affd7b7a6c0a5e4"
+    samples     = "https://koodous.com/rulesets/5862/apks"
+    author      = "Eduardo Novella"
+
+  strings:
+    // clang version 6.0.0 (Byteguard 0.6) (git@sysrepo.byted.org:dingbaozeng/native_obfuscator.git 448f20ff6eb06dd336dd81846d6a7dc8ba8c961b)
+    // Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
+    $clang_version = "clang version 6.0.0 (Byteguard 0.6) (git@sysrepo.byted.org:dingbaozeng/native_obfuscator.git "
```

Comment:
I believe due to the way the fall through generic detection works, we can drop `$clang_version` here and make this a condition of `is_elf` + `based_on`

Reply:
I like having both if you don't mind.

Reply:
So does it need both? Or just one?

If we're keeping both, imho, it should be because it needs both to properly detect this version.

If we explicitly have two samples which hit this on different strings and are the same version, then this makes sense.

Reply:
As you wish....... Done